tox.ini

```
@@ -0,0 +1,12 @@
+[tox]
+envlist =
+    py{311, 310, 39}
+skipsdist = true
+
+[testenv]
+passenv =
+    FORCE_COLOR
+deps =
+    -rrequirements.txt
+commands =
+    python -m pytest {posargs}
```

```suggestion
    python -bb -X dev -W error -m pytest {posargs}
```

Standard robust invocation to match our `make test` invocation, run in dev mode (with extra checks and debug output on error), check for `BytesWarning`, and catch any warnings prior to full Pytest initialization and its own warning filter installation.